src/appleseed/renderer/kernel/rendering/generic/tilejob.cpp

```diff
@@ -134,8 +134,9 @@ void TileJob::execute(const size_t thread_index)
             : nullptr;
 
     // Call the pre-render tile callback.
-    if (tile_callback)
-        tile_callback->on_tile_begin(&m_frame, m_tile_x, m_tile_y);
+    if (tile_callback) {
+        tile_callback->on_tile_begin(&m_frame, m_tile_x, m_tile_y, thread_index, m_tile_renderers.size());
```

Comment:
Be careful with naming, you used `nb_threads` everywhere but it's actually the number of tile renderer.

Reply:
That should be fine because there is always one tile renderer per thread. But yeah, it would be more explicit to pass the thread count.

Reply:
Should I add `thread_count` to the `TileJob` constructor and use it instead of `m_tile_renderers.size()`?

I'd then pass it to `TileJobFactory::create`, so we could use `m_thread_count` in here: 

https://github.com/appleseedhq/appleseed/blob/8053115547b9b804087586e3e8610eab42b1f6f4/src/appleseed/renderer/kernel/rendering/generic/genericframerenderer.cpp#L366-L374

Reply:
I think the way you did it (passing both thread index and thread count) is better.

Reply:
Ah sorry, I got confused:

> Should I add thread_count to the TileJob constructor and use it instead of m_tile_renderers.size()?
> I'd then pass it to TileJobFactory::create, so we could use m_thread_count in here

Yes, that sounds better indeed.

Reply:
Fixed it 🙂